ui/src/components/Entity/EntityViews.jsx

```diff
@@ -57,215 +59,345 @@ class EntityViews extends React.Component {
+    return (
+      <Tab
+        id="view"
+        icon="documentation"
+        title={<FormattedMessage id="entity.info.view" defaultMessage="View" />}
+        panel={<DocumentViewMode document={entity} activeMode={activeMode} />}
+      />
+    );
+  }
+
+  renderTextMode() {
+    const { entity, activeMode } = this.props;
+
+    if (!entity.schema.isAny(['Pages', 'Image'])) {
+      return;
+    }
+
+    return (
+      <Tab
+        id="text"
+        icon="plaintext"
+        title={<FormattedMessage id="entity.info.text" defaultMessage="Text" />}
+        panel={<DocumentViewMode document={entity} activeMode={activeMode} />}
+      />
+    );
+  }
+
+  renderBrowseMode() {
+    const { entity, children, activeMode } = this.props;
+
+    if (!entity.schema.isA('Folder')) {
+      return;
+    }
+
+    return (
+      <Tab
+        id="browse"
+        disabled={children.total < 1}
+        icon="folder"
+        title={
+          <TextLoading loading={children.isPending}>
+            {entity.schema.isA('Email') && (
+              <FormattedMessage
+                id="entity.info.attachments"
+                defaultMessage="Attachments"
               />
             )}
-          {entity?.schema?.matchable && !isPreview && (
-            <Tab
-              id="similar"
-              disabled={similar.total === 0}
-              title={
-                <TextLoading loading={similar.total === undefined}>
-                  <Icon icon="similar" className="left-icon" />
-                  <FormattedMessage
-                    id="entity.info.similar"
-                    defaultMessage="Similar"
-                  />
-                  <ResultCount result={similar} />
-                </TextLoading>
-              }
-              panel={<EntitySimilarMode entity={entity} />}
-            />
-          )}
-          {entity?.collection?.writeable && entity.schema.isA('Table') && (
-            <Tab
-              id="mapping"
-              title={
-                <>
-                  <Icon icon="new-object" className="left-icon" />
-                  <FormattedMessage
-                    id="entity.mapping.view"
-                    defaultMessage="Generate entities"
-                  />
-                </>
-              }
-              panel={<EntityMappingMode document={entity} />}
+            {!entity.schema.isA('Email') && (
+              <FormattedMessage
+                id="entity.info.documents"
+                defaultMessage="Documents"
+              />
+            )}
+            <ResultCount result={children} />
+          </TextLoading>
+        }
+        panel={<DocumentViewMode document={entity} activeMode={activeMode} />}
+      />
+    );
+  }
+
+  renderReferenceModes() {
+    const { entity, references, reference, referenceQuery, activeMode } =
+      this.props;
+
+    if (!references.total && references.isPending) {
+      return <Tab id="loading" title={<TextLoading loading={true} />} />;
+    }
+
```

Comment:
I think this is unnecessary as this is never rendered if references are still being loaded.